ConnectionFactory.php

```diff
@@ -42,22 +43,34 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        $connection = DriverManager::getConnection($params, $config, $eventManager);
-
         if (! isset($params['pdo']) && ! isset($params['charset'])) {
-            $params            = $connection->getParams();
-            $params['charset'] = 'utf8';
-            $driver            = $connection->getDriver();
+            $wrapperClass = Connection::class;
+            if (isset($params['wrapperClass'])) {
```

Comment:
Not a fan of inlining this logic here, that's what's DriverManager for. Other places using it are still having old behaviour and in case we update DriverManager, it can break ConnectionFactory, since it unsets wrapperClass option.

Reply:
I agree this should be ported to DriverManager, but fixing this here was the faster choice for Symfony users. Once this is fixed in DBAL 2.9, we can remove it here and bump the doctrine/dbal dependency.